bite/BiteDataField.cpp

```diff
@@ -7,41 +7,59 @@
 
 BiteDataField::BiteDataField(const shared_ptr<EncryptedData> &_encryptedKeyPlusData, uint64_t _epoch)
-    : encryptedData(_encryptedKeyPlusData), epoch(_epoch) {
+    : keyPlusEncryptedData(_encryptedKeyPlusData), epoch(_epoch) {
     CHECK_STATE(_encryptedKeyPlusData);
     CHECK_STATE(_encryptedKeyPlusData->size() > BITE_ENCRYPTED_AES_KEY_LEN);
     
-    
     auto aesKeyArray = std::make_shared<std::array<uint8_t, BITE_ENCRYPTED_AES_KEY_LEN>>();
     std::copy_n(_encryptedKeyPlusData->begin(), BITE_ENCRYPTED_AES_KEY_LEN, aesKeyArray->begin());
     encryptedAESKey = std::make_shared<EncryptedAESKey>(aesKeyArray);
-    serializedData = make_shared<vector<uint8_t> >();
-    serializedData->reserve(BITE_MIN_DATA_LEN + _encryptedKeyPlusData->size());
+
+    // build serialized RLP-encoded data field
     uint64_t epochBE = boost::endian::native_to_big(_epoch);
-    uint8_t* epochBytes = reinterpret_cast<uint8_t*>(&epochBE);
-    serializedData->insert(serializedData->end(), epochBytes, epochBytes + sizeof(epochBE));
-    serializedData->insert(serializedData->end(), encryptedData->begin(), encryptedData->end());
+    std::vector<uint8_t> epochBytes(reinterpret_cast<uint8_t*>(&epochBE),
+                                reinterpret_cast<uint8_t*>(&epochBE) + sizeof(epochBE));
+
+    RLPStream list;
+    list << epochBytes << *_encryptedKeyPlusData;
+
+    RLPStream listOfLists;
+    listOfLists << list;
+
+    serializedData = make_shared<vector<uint8_t> >(listOfLists.encode());
 }
 
 BiteDataField::BiteDataField(const std::shared_ptr<std::vector<uint8_t> > &_data) {
-    CHECK_STATE(_data)
-    CHECK_STATE(_data->size() > BITE_EPOCH_ID_LEN + BITE_ENCRYPTED_AES_KEY_LEN);
+    CHECK_STATE(_data);
 
-    auto keyVec = std::make_shared<std::array<uint8_t, BITE_ENCRYPTED_AES_KEY_LEN> >();
+    // parse RLP-encoded tx data field
+    RLPItem rlp(*_data);
+    CHECK_STATE2(rlp.isList(), "RLP item is not a list");
+    CHECK_STATE2(rlp.size() >= 1, "RLP item should have at least 1 item");
+    
+    // Get 1st item from list
+    RLPItem rlp0 = rlp[0];
 
-    std::copy_n(_data->begin() + BITE_EPOCH_ID_LEN, BITE_ENCRYPTED_AES_KEY_LEN, keyVec->begin());
+    CHECK_STATE2(rlp0.isList(), "RLP item 0 is not a list");
+    CHECK_STATE2(rlp0.size() == 2, "RLP item 0 should have exactly 2 fields - EPOCH_ID, and bite encrypted data");
 
-    auto encryptedDataStart = _data->begin() + BITE_EPOCH_ID_LEN;
+    // validate encrypted data
+    keyPlusEncryptedData = make_shared<std::vector<uint8_t>>(rlp0[1].asBytes());
+    CHECK_STATE2(keyPlusEncryptedData->size() >= BITE_ENCRYPTED_AES_KEY_LEN,
+        "Incorrectly formatted BITE transaction: Encrypted data size is not at least " + to_string(BITE_ENCRYPTED_AES_KEY_LEN) + " bytes, found: " + to_string(keyPlusEncryptedData->size()));
+    // get encrypted key part
+    auto keyVec = std::make_shared<std::array<uint8_t, BITE_ENCRYPTED_AES_KEY_LEN> >();
```

Comment:
can we use libbls objects here to avoid low-level bytes manipulation?
for example, 
```
keyVec = libBLS::Ciphertext::fromBytes( *keyPlusEncryptedData ).key.toBytes();
```